metrics/metrics.go

```diff
@@ -58,3 +63,55 @@ func (m *LabelMap) GetFloat(key string) *expvar.Float {
 func CurrentFDs() int {
 	return currentFDs()
 }
+
+type histogram struct {
+	// buckets is a list of bucket boundaries, in increasing order.
+	buckets []float64
+	// buckStrings is a list of the same buckets, but as strings.
+	// To avoid allocations, this is only populated on first use.
+	bucketStrings []string
+
+	lm    *LabelMap
+	sum   *expvar.Float
+	count *expvar.Int
+}
+
+// NewHistogram returns a new histogram that reports to the given
+// expvar map under the given name.
+//
+// The buckets are the boundaries of the histogram buckets, in
+// increasing order. The last bucket is +Inf.
+func NewHistogram(name string, buckets []float64) *histogram {
+	if !slices.IsSorted(buckets) {
+		panic("buckets must be sorted")
+	}
+	lm := &LabelMap{
```

Comment:
LabelMap will return keys sorted alphabetically, while prometheus metric format [requires](https://github.com/prometheus/docs/blob/main/content/docs/instrumenting/exposition_formats.md#histograms-and-summaries) buckets to appear in increasing numeric order.

Reply:
Done